src/audio/copier/copier.c

```diff
@@ -1313,7 +1315,30 @@ static int copier_params(struct comp_dev *dev, struct sof_ipc_stream_params *par
 				ret = cd->endpoint[i]->drv->ops.params(cd->endpoint[i],
 								       &demuxed_params);
 			} else {
-				ret = cd->endpoint[i]->drv->ops.params(cd->endpoint[i], params);
+				if (dev->ipc_config.type == SOF_COMP_HOST) {
+					child_dev = cd->endpoint[i];
+					ret = host_verify_params(child_dev, params);
```

Comment:
@btian1 can we call comp_verify_params() directly here?

Reply:
it is more than call comp_verify_params, let me use a new commit to make the change.

Reply:
> 

@lgirdwood , please check the commit message, we saved about 1.9mcps for a typical playback and 0x300 endpoint buffer size for each time copier with host calling.